docs/Client.md

```diff
@@ -0,0 +1,1011 @@
+
+      fastifyServerUndiciClient.request({
+        path: '/',
+        method: 'GET'
+      }).then(({ statusCode, body }) => {
+        console.log(`response received ${statusCode}`)
+        body.setEncoding('utf8')
+        body.on('data', console.log)
+
+        nodeServerUndiciClient.close()
+        fastifyServerUndiciClient.close()
+        fastifyServer.close()
+        nodeServer.close()
+      })
+    })
+})
+```
+
+### `Client.upgrade(options[, callback])`
+
+Upgrade the client to a different protocol.
+
+Arguments:
+
+* **options** `UpgradeOptions`
+  
+* **callback** `(error: Error | null, data: UpgradeData) => void` (optional)
+
+Returns: `void | Promise<UpgradeData>` - Only returns a `Promise` if no `callback` argument was passed
+
+#### Parameter: `UpgradeOptions`
+
+* **path** `string`
+* **method** `string` (optional) - Default: `'GET'`
+* **headers** `UndiciHeaders` (optional) - Default: `null`
+* **protocol** `string` (optional) - Default: `'Websocket'` - A string of comma separated protocols, in descending preference order.
+* **signal** `AbortSignal | EventEmitter | null` (optional) - Default: `null`
+
+#### Parameter: `UpgradeData`
+
+* **headers** `http.IncomingHeaders`
+* **socket** `stream.Duplex`
+* **opaque** `unknown`
+
+#### Example 1 - Basic Upgrade Request
+
+```js
+'use strict'
+const { Client } = require('undici')
+const { createServer } = require('http')
+
+const server = createServer((request, response) => {
+  response.statusCode = 101
+  response.setHeader('connection', 'upgrade')
+  response.setHeader('upgrade', request.headers.upgrade)
+  response.end()
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+
+  client
+    .upgrade({ path: '/' })
+    .then(({ headers, socket }) => {
+      socket.on('end', () => {
+        console.log(`upgrade: ${headers.upgrade}`)
+        client.close()
+        server.close()
+      })
+      socket.end()
+    })
+    .catch(error => {
+      console.error(error)
+      client.close()
+      server.close()
+    })
+})
+```
+
+## Instance Properties
+
+### `Client.busy`
+
+* `boolean`
+
+`true` if pipeline is saturated or blocked. Indicates whether dispatching further requests is meaningful.
+
+### `Client.closed`
+
+* `boolean`
+
+`true` after `client.close()` has been called.
+
+### `Client.connected`
+
+* `number`
+
+Number of acive client connections. The client will lazily create a connection when it receives a request and will destroy it if there is no activity for the duration of the `timeout` value.
+
+### `Client.destroyed`
+
+* `boolean`
+
+`true` after `client.destroyed()` has been called or `client.close()` has been called and the client shutdown has completed.
+
+### `Client.pending`
+
+* `number`
+
+Number of queued requests.
+
+### `Client.pipelining`
+
+* `number`
+
+Property to get and set the pipelining factor.
```

Comment:
Is there a more visible way to show whether a property is both get and set?

Reply:
I could move these sections to be above the _Instance Methods_ so that the user sees them first when scrolling through. And then for like getter/setter indication I'm not entirely sure.

Reply:
I'm mostly concerned about the set/get. Not a blocker.